### PR TITLE
New compose design system

### PR DESCRIPTION
**Pull request by Milan-Cerovsky**
# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [x] Update **documentation** as appropriate (e.g [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), [**CHANGELOG.md**](../blob/main/CHANGELOG.md), etc.)
- [x] **Run the app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

> **Note**
> It is good practice to provide before and after UI  **screenshots** in the description of this PR. This is only applicable for changes that modify the UI.

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [x] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [x] Review the **documentation**, [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), etc. as appropriate
- [x] **Run the app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: docs/Design System Generation.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+The design system contains dimensions, colors and color modes in Figma. Because the amount of these variables makes it 
+hard to reflect the changes manually it is a better use of time to automate it.
+
+## AI Automation
+
+The idea is to output the variables into a json and then have AI (i.e. chatgpt) generate kotlin code from it.
+
+## Figma variables into json
+
+Use 'Variables Exporter For Dev Mode' figma plugin. It outputs all variables into a simple structured json.
```
Comment (HonzaR) [inline docs/Design System Generation.md:12]: Can we add a reference to the tool's homepage?

**Review by HonzaR — COMMENTED** [review]
Looks good to me! I'm sharing a few first thoughts.

File: docs/Design System Generation.md (hunk 1 above)
Comment (Milan-Cerovsky) [reply docs/Design System Generation.md]: done

File: docs/Design System Generation.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+- the more descriptive and exhaustive the prompt is the better
+
+Example Prompt:
+
+```
+System:
+You are an expert Android Developer and you are able to understand kotlin & json.
+
+Task:
+Your task is to convert convert given json into kotlin code. In the next message you will be given a json file and
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply docs/Design System Generation.md]: done

File: docs/Design System Generation.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Notes:
+- always output variables in kotlin compatible way
+- examples: 'button-background'='buttonBackground' 
+```
+
+# AI Prompt for light theme
+
+```
+Populate this data structure by the colors from give json. Output a variable 'val LightZashiColorModes =
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply docs/Design System Generation.md]: done

File: docs/Design System Generation.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+```
+Populate this data structure by the colors from give json. Output a variable 'val LightZashiColorModes = 
+ZashiColorModes(...)'.
+```
+
+# AI Prompt for dark theme
+
+```
+Populate this data structure by the colors from give json. Output a variable 'val DarkZashiColorModes =
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply docs/Design System Generation.md]: done

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/dimensions/ZashiDimensions.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+package co.electriccoin.zcash.ui.design.theme.dimensions
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.staticCompositionLocalOf
+
+val ZashiDimensions: ZashiDimensionsInternal
+    @Composable get() = LocalZashiDimensions.current
+
+@Suppress("CompositionLocalAllowlist")
+internal val LocalZashiDimensions = staticCompositionLocalOf<ZashiDimensionsInternal> { error("no colors specified") }
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/dimensions/ZashiDimensions.kt]: done

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/typography/ZashiTypography.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+package co.electriccoin.zcash.ui.design.theme.typography
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.staticCompositionLocalOf
+
+val ZashiTypography: ZashiTypographyInternal
+    @Composable get() = LocalZashiTypography.current
+
+@Suppress("CompositionLocalAllowlist")
+internal val LocalZashiTypography = staticCompositionLocalOf<ZashiTypographyInternal> { error("no colors specified") }
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/typography/ZashiTypography.kt]: done

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/dimensions/ZashiDimensions.kt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+@file:Suppress("MagicNumber")
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/dimensions/ZashiDimensions.kt]: done

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/typography/ZashiTypography.kt [hunk 8]
```diff
@@ -0,0 +1,12 @@
+@file:Suppress("MagicNumber")
```
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/typography/ZashiTypography.kt:1]: done

**Comment by Milan-Cerovsky** [comment]
No task actually for this actually

**Comment by Milan-Cerovsky** [comment]
@HonzaR using underscores to color names i.e. Gray._50 throws a ktlint format exception and I cannot suppress it

**Review by HonzaR — APPROVED** [review]
Great! Thank you for these changes.

**Pull request merged**